runtime/wasm/src/host.rs

```diff
@@ -179,45 +179,40 @@ impl RuntimeHost {
                 store: store.clone(),
             };
 
-            // Start the mapping as a WASM module
-            let mut module = WasmiModule::new(&module_logger, wasmi_config, task_sender)
-                .expect("Failed to load module");
+            // Validate the mapping's WASM module
+            let valid_module = ValidModule::new(&module_logger, wasmi_config, task_sender)
+                .expect("Failed to validate module");
```

Comment:
If the subgraph was not compiled with our toolchain, it might happen. But note that this is directly inside `spawn`, so panicking is a reasonable way to fail. Dropping `handle_event_receiver` will cause the failure to propagate up, ultimately causing `process_log` to fail and the subgraph status to be set as failed. I've added a comment clarifying this.

This will happen the next time the subgraph tries to process an event, so it will not be immediate. It would be more complex to make it immediate and I'm not sure it would be desirable.

Reply:
Ah yes, it's inside the spawn and we have panic handling for the runtime threads in place. Great. This makes sense now. Thanks for clarifying!